src/track/beatfactory.cpp

```diff
@@ -95,6 +95,14 @@ mixxx::BeatsPointer BeatFactory::makePreferredBeats(
                 sampleRate, subVersion, constBPM, firstBeat * 2);
         return pGrid;
     } else if (version == BEAT_MAP_VERSION) {
+        if (beats.size() > 48) {
```

Comment:
I think this unrelated and slipped in by mistake.

Reply:
Oh yes.

Reply:
Forced pushed without that change.